packages/jest-matchers/src/__tests__/matchers-test.js

```diff
@@ -92,6 +92,44 @@ describe('.toBeNaN()', () => {
   });
 });
 
+describe('.toBeNull()', () => {
+  [{}, [], true, 1, 'a', 0.5, new Map(), () => {}, Infinity].forEach(v => {
+    test(`fails for '${stringify(v)}' with .not`, () => {
+      jestExpect(v).not.toBeNull();
+      expect(() => jestExpect(v).toBeNull())
+        .toThrowError(/expected.*to be null/);
+    });
+  });
+
+  it('pass for null', () => {
+    jestExpect(null).toBeNull();
+    expect(() => jestExpect(null).not.toBeNull())
+      .toThrowError(/expected.*not to be null/);
+  });
+});
+
+describe('.toBeDefined(), .toBeUndefined()', () => {
+  [{}, [], true, 1, 'a', 0.5, new Map(), () => {}, Infinity].forEach(v => {
+    test(`'${stringify(v)}' is defined`, () => {
+      jestExpect(v).toBeDefined();
+      jestExpect(v).not.toBeUndefined();
+      expect(() => jestExpect(v).not.toBeDefined())
+        .toThrowError(/not to be defined/);
+      expect(() => jestExpect(v).toBeUndefined())
+        .toThrowError(/undefined/);
```

Comment:
can you add a little more text to this regex?
there's a hight change of false green if you just match it to `/undefined/` :)